hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -129,7 +129,17 @@ class HoodieCatalogTable(val spark: SparkSession, var table: CatalogTable) exten
   /**
    * Table schema
    */
-  lazy val tableSchema: StructType = table.schema
+  lazy val tableSchema: StructType = if (table.schema.nonEmpty) {
```

Comment:
I prefer to call `loadTableSchemaByMetaClient` directly without having to decide if `table.schema` is empty.
Parsing the metadata on spark does not need to depend on the meta from metastore.

Reply:
@YannByron The `table.schema` is not obtained from the hive metastore, but is specified by the user. Only when the user does not specify a schema, we can parse and obtain the schema from the file through the method `loadTableSchemaByMetaClient`